src/ray/gcs/client_def.h

```diff
@@ -0,0 +1,95 @@
+#ifndef RAY_GCS_CLIENT_DEF_H
```

Comment:
Maybe just define `ClientOption` in `gcs_client.h`. the name `client_def.h` is kind of ambiguous.